File: backend/core/src/main/java/org/demo/core/model/entity/KnowledgeBase.java
package org.demo.core.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 知识库表实体类
 * 功能: 存储知识库的基本信息，支持分级设计
 * 关联用户故事: US-006, US-008, US-010
 */
@Data
@TableName(value = "knowledge_base", autoResultMap = true)
public class KnowledgeBase {

    /**
     * 知识库唯一标识
     */
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private String id;

    /**
     * UUID标识
     */
    @TableField("uuid")
    private String uuid;

    /**
     * 知识库名称（必填）
     */
    @TableField("name")
    private String name;

    /**
     * 知识库描述
     */
    @TableField("description")
    private String description;

    /**
     * 知识库图标URL
     */
    @TableField("icon")
    private String icon;

    /**
     * 作用域类型（system/school/course/agent/personal）
     */
    @TableField("scope_type")
    private String scopeType;

    /**
     * 作用域ID
     */
    @TableField("scope_id")
    private Integer scopeId;

    /**
     * 父知识库ID
     */
    @TableField("parent_kb_id")
    private String parentKbId;

    /**
     * 创建者ID
     */
    @TableField("owner_id")
    private String ownerId;

    /**
     * 创建者ID（别名字段）
     */
    @TableField("user_id")
    private String userId;

    /**
     * 访问级别（public/protected/private）
     */
    @TableField("access_level")
    private String accessLevel;

    /**
     * 文档数量
     */
    @TableField("document_count")
    private Integer documentCount;

    /**
     * 分块数量
     */
    @TableField("chunk_count")
    private Integer chunkCount;

    /**
     * 总文件大小（字节）
     */
    @TableField("total_size")
    private Long totalSize;

    /**
     * 分块大小
     */
    @TableField("chunk_size")
    private Integer chunkSize;

    /**
     * 分块重叠
     */
    @TableField("chunk_overlap")
    private Integer chunkOverlap;

    /**
     * 向量模型（如text-embedding-3）
     */
    @TableField("embedding_model")
    private String embeddingModel;

    /**
     * 向量模型ID（外键）
     */
    @TableField("embedding_model_id")
    private String embeddingModelId;

    /**
     * 检索配置
     */
    @TableField(value = "retrieval_config", typeHandler = JacksonTypeHandler.class)
    private Object retrievalConfig;

    /**
     * 创建时间
     */
    @TableField("create_time")
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField("update_time")
    private LocalDateTime updateTime;
}

File: frontend/src/api/knowledgeBase.ts
import { http } from '@/utils/http'

// 知识库相关类型定义
export interface KnowledgeBase {
  id: string
  uuid: string
  name: string
  description?: string
  icon?: string
  scopeType: string
  scopeId?: number
  parentKbId?: string
  ownerId: string
  accessLevel: string
  documentCount: number
  chunkCount: number
  totalSize: number
  chunkSize: number
  chunkOverlap: number
  embeddingModel?: string
  embeddingModelId?: string
  retrievalConfig?: {
    top_k?: number
    similarity_threshold?: number
    max_context_length?: number
  }
  createTime: string
  updateTime: string
}

export interface KnowledgeBaseCreateDTO {
  name: string
  description?: string
  icon?: string
  scopeType: string
  scopeId?: number
  parentKbId?: string
  accessLevel?: string
  chunkSize?: number
  chunkOverlap?: number
  embeddingModelId?: string
  retrievalConfig?: {
    top_k?: number
    similarity_threshold?: number
    max_context_length?: number
  }
}

export interface KnowledgeBasePatchDTO {
  name?: string
  description?: string
  icon?: string
  accessLevel?: string
  chunkSize?: number
  chunkOverlap?: number
  embeddingModelId?: string
  retrievalConfig?: {
    top_k?: number
    similarity_threshold?: number
    max_context_length?: number
  }
}

export interface Document {
  uuid: string
  name: string
  filename: string
  fileUrl?: string
  fileSize: number
  fileType: string
  chunkCount?: number
  status: 'uploading' | 'processing' | 'processed' | 'failed'
  errorMessage?: string
  processedAt?: string
  knowledgeBaseId: string
  userId: string
  createdAt: string
  updatedAt: string
}

export interface PageResult<T> {
  list: T[]
  total: number
}

export interface RagQueryResultItem {
  vector_id: number
  chunk_index: number
  score: number
  content: string
}

export interface RagQueryResult {
  result_num: number
  results: RagQueryResultItem[]
}

/**
 * 创建知识库
 */
export const createKnowledgeBase = async (data: KnowledgeBaseCreateDTO): Promise<KnowledgeBase> => {
  return http.post('/v1/knowledge-bases', data)
}

/**
 * 获取知识库列表（支持分页、搜索、筛选）
 */
export const getKnowledgeBaseList = async (params?: {
  page?: number
  pageSize?: number
  search?: string
  scopeType?: string
  accessLevel?: string
}): Promise<PageResult<KnowledgeBase>> => {
  return http.get('/v1/knowledge-bases', { params })
}

/**
 * 获取知识库详情
 */
export const getKnowledgeBase = async (uuid: string): Promise<KnowledgeBase> => {
  return http.get(`/v1/knowledge-bases/${uuid}`)
}

/**
 * 更新知识库（部分更新）
 */
export const updateKnowledgeBase = async (
  uuid: string,
  data: KnowledgeBasePatchDTO
): Promise<KnowledgeBase> => {
  return http.patch(`/v1/knowledge-bases/${uuid}`, data)
}

/**
 * 删除知识库
 */
export const deleteKnowledgeBase = async (uuid: string): Promise<void> => {
  return http.delete(`/v1/knowledge-bases/${uuid}`)
}

/**
 * 获取知识库的文档列表
 */
export const getDocumentList = async (
  kbUuid: string,
  params?: {
    page?: number
    pageSize?: number
    status?: string
  }
): Promise<PageResult<Document>> => {
  return http.get(`/v1/knowledge-bases/${kbUuid}/documents`, { params })
}

/**
 * 上传文档到知识库
 */
export const uploadDocument = async (
  kbUuid: string,
  file: File,
  onUploadProgress?: (progressEvent: any) => void
): Promise<Document> => {
  const formData = new FormData()
  formData.append('file', file)

  return http.post(`/v1/knowledge-bases/${kbUuid}/documents`, formData, {
    headers: {
      'Content-Type': 'multipart/form-data'
    },
    onUploadProgress
  })
}

/**
 * 获取文档详情
 */
export const getDocument = async (uuid: string): Promise<Document> => {
  return http.get(`/v1/documents/${uuid}`)
}

/**
 * 删除文档
 */
export const deleteDocument = async (uuid: string): Promise<void> => {
  return http.delete(`/v1/documents/${uuid}`)
}

/**
 * 查询知识库
 */
export const queryKnowledgeBase = async (
  kbUuid: string,
  kb_id: string,
  query: string,
  top_k: number = 5,
  similarity_threshold: number = 0.5): Promise<RagQueryResult> => {
  return http.post(`/v1/knowledge-bases/${kbUuid}/query`, {
    knowledge_base_id: kb_id,
    query: query,
    top_k: top_k,
    similarity_threshold: similarity_threshold
  })
}
